### PR TITLE
1187 deps upgrade

### LINKED ISSUE #1187: Deps upgrade
Time to update dependencies. Yay.

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #1187. uh.. sorry? 🤷  😬 

Well, that sucked. Let's not do this again 🙄 

The tests are broken.. but so am I. Will make a new issue to fix them up.

### CODE REVIEW

File: packages/common/src/api/GqlContext.tsx [hunk 1]
```diff
@@ -54,25 +61,19 @@ class GQLClient extends GraphQLClient {
...
-  }
-
   public request<T, V = Variables>(
-    document: RequestDocument,
+    documentOrOptions: RequestDocument | RequestOptions<V>,
     variables?: V,
     requestHeaders?: RequestInit['headers']
   ): Promise<T> {
-    const response = this.client.request(document, variables, requestHeaders);
+    const options = documentOrOptions as RequestOptions<V>;
```
Comment (mark-prins) [inline packages/common/src/api/GqlContext.tsx:69]: quick monkey around to get typescript to be happy again.

File: packages/common/src/api/GqlContext.tsx [hunk 2]
```diff
@@ -123,7 +119,10 @@ interface ApiProviderProps {
   url: string;
 }
 
-export const GqlProvider: FC<ApiProviderProps> = ({ url, children }) => {
+export const GqlProvider: FC<PropsWithChildren<ApiProviderProps>> = ({
```
Comment (mark-prins) [inline packages/common/src/api/GqlContext.tsx:122]: and a _lot_ of these. React now requires explicit declaration of support for `children` ( previously using `React.FC` included a children prop )

File: packages/common/src/intl/locales/en/app.json [hunk 3]
```diff
@@ -51,7 +51,7 @@
   "cmdk.placeholder": "Type a command or search",
   "login.heading": "Simple.\nPowerful.\nPharmaceutical\nManagement.",
   "login.body": "Whether you run a pharmaceutical distribution warehouse, or are a manufacturer needing dozens of connected users, or a small dispensary, mSupply is a valuable partner.",
-  "login.store-changed": "Welcome to the {{store, string}} store",
+  "login.store-changed": "Welcome to the {{store}} store",
```
Comment (mark-prins) [inline packages/common/src/intl/locales/en/app.json:54]: this was throwing a warning

File: packages/common/src/ui/components/loading/BouncingGuy/BouncingGuy.tsx [hunk 4]
```diff
@@ -1,8 +1,7 @@
 import React, { FC } from 'react';
-import { GlobalStyles, Typography } from '@mui/material';
+import { GlobalStyles } from '@mui/material';
```
Comment (mark-prins) [inline packages/common/src/ui/components/loading/BouncingGuy/BouncingGuy.tsx:2]: this made me sad. The use of the translation in here meant that the `Suspense` was always suspended. not an issue previously.. but it is now. after a lot of time getting to the root of it.. just removed the translation and everything is fine 🙄

File: packages/common/src/ui/components/portals/DetailPanel/DetailPanel.stories.tsx [hunk 5]
```diff
@@ -27,7 +27,7 @@ const Template: Story<DetailPanelArgs> = args => {
   const { Actions, Sections } = args;
 
   return (
-    <StoryProvider locale="en">
+    <StoryProvider>
```
Comment (mark-prins) [inline packages/common/src/ui/components/portals/DetailPanel/DetailPanel.stories.tsx:30]: wasn't using the `locale`

File: packages/host/src/components/Login/LoginLayout.tsx [hunk 6]
```diff
@@ -35,7 +36,7 @@ export const LoginLayout = ({
       <Box
         flex="1 0 50%"
         sx={{
-          backgroundImage: theme => theme.mixins.gradient.primary,
+          backgroundImage: (theme: Theme) => theme.mixins.gradient.primary,
```
Comment (mark-prins) [inline packages/host/src/components/Login/LoginLayout.tsx:39]: not sure why this is complaining now.. but fixed

File: packages/host/tsconfig.json [hunk 7]
```diff
@@ -1,3 +1,10 @@
 {
-  "extends": "../../tsconfig.json"
+  "extends": "../../tsconfig.json",
+  "compilerOptions": {
```
Comment (mark-prins) [inline packages/host/tsconfig.json:3]: hm. yes. kbar isn't ready for react 18 yet.. at least as far as typescript is concerned. I've had to temporarily add in some override typings.

File: packages/host/webpack.config.js [hunk 8]
```diff
@@ -1,7 +1,7 @@
 const ReactRefreshWebpackPlugin = require('@pmmmwh/react-refresh-webpack-plugin');
 const CopyPlugin = require('copy-webpack-plugin');
 const webpack = require('webpack');
-const HotModuleReplacementPlugin = webpack.HotModuleReplacementPlugin;
```
Comment (mark-prins) [inline packages/host/webpack.config.js:4]: as per the warning when run: HMR is automatically added when `hot: true` is used

File: packages/inventory/src/Stocktake/api/hooks.ts [hunk 9]
```diff
@@ -182,9 +182,9 @@ export const useStocktakeItems = (): UseQueryResult<StocktakeSummaryItem[]> => {
       ([itemId, lines]) => {
         return {
           id: itemId,
-          item: lines[0].item,
+          item: lines[0]?.item,
```
Comment (mark-prins) [inline packages/inventory/src/Stocktake/api/hooks.ts:185]: I had to remove the `[T, ...T[]]` type syntax and as a result, cope with potentially undefined array elements.
Once again, not sure why ts is unhappy now, but couldn't find an alternative workaround

**Review by clemens-msupply — APPROVED** [review]
Thanks! only skimmed through it. Had a look at your comments but could add anything helpful to it.

**Pull request merged**